synapseclient/models/mixins/table_operator.py

```diff
@@ -2428,63 +2799,249 @@ async def main():
+            # will allow for the purge cache function to clean up files that
+            # end up getting left here. It is also to account for the fact that
+            # the temp directory may not have enough disk space to hold a file
+            # we need to upload (As is the case on EC2 instances)
+            temp_dir = client.cache.get_cache_dir(file_handle_id=111111111)
+            os.makedirs(temp_dir, exist_ok=True)
+            with tempfile.NamedTemporaryFile(
+                prefix="chunked_csv_for_synapse_store_rows",
+                suffix=".csv",
+                dir=temp_dir,
+            ) as temp_file:
+                try:
+                    # TODO: This portion of the code should be updated to support uploading a file from memory using BytesIO (Ticket to be created)
+                    # TODO: The way that the dataframe is split up is also needed. As of right now we are writing the CSV, and then letting the CSV
+                    # TODO: upload process take over. However, if we are uploading from memory, then we don't have the ability to write to a file, and then
+                    # TODO: upload that file. We need to be able to split the dataframe into chunks, and then upload those chunks to Synapse.
+                    values.to_csv(
+                        temp_file.name,
+                        index=False,
+                        float_format="%.12g",
+                        **(to_csv_kwargs or {}),
+                    )
+                    # NOTE: reason for flat_format='%.12g':
+                    # pandas automatically converts int columns into float64 columns when some cells in the column have no
+                    # value. If we write the whole number back as a decimal (e.g. '3.0'), Synapse complains that we are writing
+                    # a float into a INTEGER(synapse table type) column. Using the 'g' will strip off '.0' from whole number
+                    # values. pandas by default (with no float_format parameter) seems to keep 12 values after decimal, so we
+                    # use '%.12g'.c
+                    # see SYNPY-267.
+
+                    with logging_redirect_tqdm(loggers=[client.logger]):
+                        await self._chunk_and_upload_csv(
+                            path_to_csv=temp_file.name,
+                            insert_size_byte=insert_size_byte,
+                            csv_table_descriptor=csv_table_descriptor,
+                            schema_change_request=schema_change_request,
+                            client=client,
+                            additional_changes=additional_changes,
+                            job_timeout=job_timeout,
+                        )
+                finally:
+                    temp_file.close()
+
+        else:
+            raise ValueError(
+                "Don't know how to make tables from values of type %s." % type(values)
             )
+
+    async def _chunk_and_upload_csv(
```

Comment:
Should this function have its own docstring (separate from `_send_update`)?

Reply:
I completed this in https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1168